src/App.tsx

```diff
@@ -1,26 +1,268 @@
+    };
+
+    try {
+      setTempTodoTitle(title);
+
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
+      setErrorMessage(ErrorMessages.ENABLE_TO_ADD_TODO);
+      throw new Error(ErrorMessages.ENABLE_TO_ADD_TODO);
+    } finally {
+      setTempTodoTitle(null);
+    }
+  };
+
+  const handleDeleteTodo = async (id: number) => {
+    try {
+      setIdsProccesing([id]);
+      await deleteTodo(id);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+    } catch {
+      setErrorMessage(ErrorMessages.ENABLE_TO_DELETE_TODO);
+    } finally {
+      setIdsProccesing([]);
+    }
+  };
+
+  const handleEditTodo = async (id: number, data: Partial<Todo>) => {
+    try {
+      setIdsProccesing([id]);
+      const editedTodo = await patchTodo(id, data);
+
+      setTodos(currentTodos =>
+        currentTodos.map(todo => {
+          if (todo.id === id) {
+            return editedTodo;
+          }
+
+          return todo;
+        }),
+      );
+    } catch {
+      setErrorMessage(ErrorMessages.ENABLE_TO_UPDATE_TODO);
+      throw new Error(ErrorMessages.ENABLE_TO_UPDATE_TODO);
+    } finally {
+      setIdsProccesing([]);
+    }
+  };
+
+  const clearCompletedTodos = async () => {
+    const filteredTodos = todos.filter(todo => todo.completed);
+    const completedIds = filteredTodos.map(todo => todo.id);
+
+    setIdsProccesing(completedIds);
+
+    try {
+      const deleteCallback = async (todo: Todo) => {
+        try {
+          await deleteTodo(todo.id);
+
+          return { id: todo.id, status: 'resolved' };
+        } catch {
+          setErrorMessage(ErrorMessages.ENABLE_TO_DELETE_TODO);
+
+          return { id: todo.id, status: 'rejected' };
+        } finally {
+          setIdsProccesing([]);
+        }
+      };
+
+      const res = await Promise.allSettled(filteredTodos.map(deleteCallback));
+
+      const resolvedIds = res.reduce(
+        (acc, item) => {
+          if (item.status === 'rejected') {
+            return acc;
+          }
+
+          if (item.value.status === 'resolved') {
+            return { ...acc, [item.value.id]: item.value.id };
+          }
+
+          return acc;
+        },
+        {} as Record<number, number>,
+      );
+
+      setTodos(currentTodos =>
+        currentTodos.filter(todo => {
+          if (resolvedIds[todo.id] && todo.completed) {
+            return false;
+          }
+
+          return true;
+        }),
+      );
+    } catch {
+      setErrorMessage(ErrorMessages.ENABLE_TO_CLEAR_COMPLETED_TODO);
+    }
+  };
+
+  const filterTodos = useMemo(() => {
+    let filteredTodos = todos;
+
+    switch (filter) {
+      case Filter.Active:
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+      case Filter.Completed:
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+      default:
+        break;
+    }
```

Comment:
move to separate file

Reply:
I see that you moved it to a separate file so let's remove it from here and use imported one